File: src/1760. Minimum Limit of Balls in a Bag/1760.py
from typing import List
class Solution:
    def minimumSize(self, nums: List[int], maxOperations: int) -> int:
        left, right, ans = 1, max(nums), 0
        
        while left <= right:
            y = (left + right) // 2
            ops = 0
            for x in nums:
                ops += ((x - 1) // y)

            # ops = sum((x - 1) // y for x in nums)
            if ops <= maxOperations:
                ans = y
                right = y - 1
            else:
                left = y + 1
        return ans
if __name__ == "__main__":
    nums = [2,4,8,2]
    maxOperations = 4
    print(Solution().minimumSize(nums, maxOperations))

File: src/1760. Minimum Limit of Balls in a Bag/1760.go
func minimumSize(nums []int, maxOperations int) int {
	max := 0
	for _, x := range nums {
		if x > max {
			max = x
		}
	}
	return sort.Search(max, func(y int) bool {
		if y == 0 {
			return false
		}
		ops := 0
		for _, x := range nums {
			ops += (x - 1) / y
		}
		return ops <= maxOperations
	})
}

